lib/broadway_sqs/ex_aws_client.ex

```diff
@@ -37,20 +37,40 @@ defmodule BroadwaySQS.ExAwsClient do
   def ack(ack_ref, successful, failed) do
     ack_options = :persistent_term.get(ack_ref)
 
-    messages =
+    messages_to_delete =
       Enum.filter(successful, &ack?(&1, ack_options, :on_success)) ++
         Enum.filter(failed, &ack?(&1, ack_options, :on_failure))
 
-    messages
+    messages_to_nack_with_timeout =
+      collect_messages_to_nack(successful, ack_options, :on_success) ++
+        collect_messages_to_nack(failed, ack_options, :on_failure)
```

Comment:
Can we write it in the same style as above for consistency?

```suggestion
      Enum.flat_map(successful, &nack(&1, ack_options, :on_success)) ++
        Enum.flat_map(failed, &nack(&1, ack_options, :on_failure))
```

And then `nack` either returns an empty list or a single element list with message and timeout. :)